documentation/nat-service/nat-service.md

```diff
@@ -0,0 +1,206 @@
+
+### 2. DNAT scenario
+
+#### 2.1 Inbound packet proccessing pipeline
+
+![dnat-inbound-packet-processing-pipe](images/Figure4-dnat-inbound-pipeline.png)
+*Figure 4 - DNAT inbound packet processing pipeline*
+
+* The pipeline determines the **inbound direction** based on the absence of a VXLAN header in the incoming packet.
+
+* A search of the inbound flow table is initiated. If a match is found, the packet is processed through the **fast path**(H/W flow table actions obtained and applied). If no match is found, the packet is handled through the **slow path**. A search for a DNAT rule is then initiated, if found, it replaces the public destination IP address and port number with the mapped VM IP address and port number.
+
+* The pipeline also initiates an VTEP lookup to obtain the VTEP IP of the DNAT mapped VM, which is then used to create an inbound flow table entry.
+
+* Both an outbound flow table entry and an inbound flow table entry are added to corresponding flow tables.
+
+* A search of the inbound flow table is triggered once the packet is re-injected. The packet will match the inbound flow table entry created previously and the flow action will be applied, which includes replacing the destination IP address/port number and adding VXLAN encapsulation.
+
+* A routing lookup is performed based on destination VTEP IP address before sending the VXLAN encapsulated packet to VM.
+
+### 2.2 Outbound packet processing pipeline
+
+![dnat-outbound-packet-processing-pipeline](images/Figure5-dnat-outbound-pipeline.png)
+*Figure 5 - DNAT outbound packet processing pipeline*
+
+* As described in section 2.1, the outbound flow table entry has been created during the handling of the *first packet* processing, when the responding packet arrives from VM, it matches the entry and performs the specified action of replacing the source IP address/port number and removing the VXLAN header before being sent to the internet.
+
+* Then it routes the packet based on DIP of the decapsulated packet.
+
+* Counting and/or metering is done, and they are optional based on the Vnet user's billing mode.
+
+● The packet is sent to Internet via the selected interface.
+
+## Configuration example
+
+``` JSON
+
+/* Define Vnet1 */
+DASH_VNET:Vnet1 {
+    "vni": 12345
+}
+
+/* Define Vnet2 */
+DASH_VNET:Vnet2 {
+    "vni": 45678
+}
+
+/* Define routing types */
+DASH_ROUTING_TYPE:vnet_encap: [
+    {
+         "name": "action1",
+         "action_type: "staticvxlan",
+         "encap_type" "vxlan"
+    }
+]
+
+/* Define Overlay routing tables */
+DASH_OVERLAY_ROUTING_TABLE:Vnet1:10.1.0.10 {
+    "routing_type":"vnet_encap", 
+    "dst_vtep":[30.0.0.22]
+    "mac_address":002244AABBCC
+},
+DASH_OVERLAY_ROUTING_TABLE:Vnet1:10.1.0.2 {
```

Comment:
Is this a new Table?

Reply:
Yes, DASH_OVERLAY_ROUTING_TABLE a new table. 

DASH_OVERLAY_ROUTING_TABLE:Vnet1:10.1.0.10 {
    "routing_type":"vnet_encap", 
    "dst_vtep":[30.0.0.22]                               # <--- a list of VTEP
    "mac_address":002244AABBCC
},

Reply:
can you please rename to DASH_VNET_ROUTE_TABLE to be aligned with existing naming convention?